test/gpu/native/studies/sort/radixSortGpu.chpl

```diff
@@ -4,425 +4,40 @@ module RadixSort {
-      // writeln(" Count: ", thisCounts);
-      // This works because counts array is all 0s and
-      // we're only adding to it. The parallelCount function leaves the
-      // rest of the array untouched (i.e. 0s) so for 2 GPUs the add might be:
-      // counts1 : 1 0 | 3 0 | 2 0 | 32 0
-      // counts2 : 0 3 | 0 2 | 0 12 | 0 2
-      // counts      : 1 3 | 3 2 | 2 12 | 32 2
-      // I think this is not a race condition for the same reason
-      firstRunReset = false; // This is a hacky solution
-
-
-      // In an ideal world I would like to use a + reduce intent here such that each iteration gets it's own
-      // copy of the counts array and it is accumulated into the main counts array
-      // And the loop should be a coforall instead of a foreach
-      // But I could not get that to work/ don't know how that works
-    }
-  }
-
-  proc parallelArrScan(ref arr: [] uint){
-    // Divide up the array into chunks of a reasonable size
-    // Config this with a scanChunkSize
-    const numScanChunks = Math.divCeil(arr.size,scanChunkSize);
-
-    if numScanChunks == 1 {
-      arrScan(arr);
-      return;
-    }
-
-    // Allocate an accumulate array
-    var scanArr : [0..<numScanChunks] uint;
-
-    var host = here;
-
-    on here.gpus[useGpuId]{
-      var gpuScanArr = scanArr;
-      var gpuArr = arr;
-      const ceil = gpuArr.domain.high;
-      // In parallel: For each chunk we do an in lane serial scan
-      @assertOnGpu
-      foreach chunk in 0..<numScanChunks {
-        const start = chunk*scanChunkSize;
-        var end = start+scanChunkSize-1;
-        if end > ceil then end = ceil;
-        gpuScanArr[chunk] = gpuArr[end]; // Save the last element before the scan overwrites it
-        serialScan(gpuArr, start, end); // Exclusive scan in serial
-        gpuScanArr[chunk] += gpuArr[end]; // Save inclusive scan in the scan Arr
-
-      }
-
-      on host {
-        // Scan the scanArr
-        // This is risky because gpuScanArr lives on the GPU
-        // 2 out of 3 scan algorithms also work on GPU
-        // So it's fine for those
-        // But the serialScan is on the CPU
-        // therefor we make sure we don't use serialScan here
-        assert(scanType!="serialScan");
-        arrScan(gpuScanArr); // funky words
-      }
-
-      @assertOnGpu
-      foreach i in 0..<gpuArr.size {
-        // In propagate the right values from scanArr
-        // to complete the global scan
-        const offset : int = i / scanChunkSize;
-        gpuArr[i] += gpuScanArr[offset];
-      }
-      arr = gpuArr;
-    }
-  }
-
-  proc arrCountAndScan(ref count: [?D] uint, ref inputArr: [] uint, exp: int, bitMask: int){
+  private proc arrCountAndScan(ref count: [?D] uint, ref inputArr: [] uint, exp: int, bitMask: int){
     // Count the number of occurrences of each bitsAtATime-bit chunk in the array
     // at the current exp position
-    on here.gpus[useGpuId]{
+    on here.gpus[0]{
```

Comment:
I may be misremembering, but I think you added this `useGpuId` to avoid device 0 on chapdl when it seems to be behaving erratically. Should we keep it for similar reasons?

Reply:
We can keep that sort of pattern. I got rid of it since the issues on chapdl seem to have resolved and there wasn't a need for it anymore. It should not be too hard to change it back.

Reply:
I don't want to overstate my stance. It is up to you really. Almost all of our testing uses `here.gpus[0]`, anyways.

Reply:
I've re-added `useGpuId`